src/setFocusHandler.js

```diff
@@ -9,7 +9,7 @@ const onWindowFocus = () => {
   const { refetchAllOnWindowFocus } = defaultConfigRef.current
 
   if (isDocumentVisible() && isOnline()) {
-    queryCache
+    defaultQueryCacheRef.current
```

Comment:
Hmmm. Does this feel funny to you? What if a user had multiple query caches in their app? How would they hook up to this? Should multiple caches be allowed?

Reply:
yes it definitely felt funny! But as you can set up the focus handler outside of any sort of react context, it's very hard to determine which cache to use. I used the same pattern as the config where there is a `defaultConfigRef` that the provider writes over. the focus handler actually uses the config ref as well so it's already a bit of a grey area...

Reply:
for reference https://github.com/tannerlinsley/react-query/blob/master/src/setFocusHandler.js#L9

Definitely open to cleaner solutions!

Reply:
Since this specific config value is "global", it kind of do make sense to use the latest one. For caches, maybe we should keep track of _all active_ caches and update them all? I think that should make it possible to get rid of `defaultQueryCacheRef`?

Reply:
I've removed the `defaultQueryCacheRef` in favour of a list of active caches